src/main/java/ch/njol/skript/expressions/ExprWeather.java

```diff
@@ -12,76 +12,92 @@
 import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
 import org.bukkit.World;
+import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
 import org.bukkit.event.weather.ThunderChangeEvent;
 import org.bukkit.event.weather.WeatherChangeEvent;
 import org.bukkit.event.weather.WeatherEvent;
 import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Weather")
-@Description("The weather in the given or the current world.")
-@Examples({"set weather to clear",
-		"weather in \"world\" is rainy"})
+@Description({
+	"The weather of a world or player.",
+	"Clearing or resetting the weather of a player will make the player's weather match the weather of the world.",
+	"Clearing or resetting the weather of a world will make the weather clear."
+})
+@Examples({
+	"set weather to clear",
+	"weather in \"world\" is rainy",
+	"reset custom weather of player",
+	"set weather of player to clear"
+})
 @Since("1.0")
 @Events("weather change")
-public class ExprWeather extends PropertyExpression<World, WeatherType> {
+public class ExprWeather extends PropertyExpression<Object, WeatherType> {
+
 	static {
-		Skript.registerExpression(ExprWeather.class, WeatherType.class, ExpressionType.PROPERTY, "[the] weather [(in|of) %worlds%]", "%worlds%'[s] weather");
+		Skript.registerExpression(ExprWeather.class, WeatherType.class, ExpressionType.PROPERTY,
+			"[the] [custom|client] weather [(in|of) %worlds/players%]",
+			"%worlds/players%'[s] [custom|client] weather");
```

Comment:
Make players come first to avoid converting player to world.
```suggestion
				"[the] [custom|client] weather [(in|of) %players/worlds%]",
				"%players/worlds%'[s] [custom|client] weather");
```